tools/fs/files.js

```diff
@@ -737,6 +745,32 @@ files.extractTarGz = function (buffer, destPath, options) {
   }
 };
 
+function waitFSToSync(dir) {
```

Comment:
I think another name for this function is in order too, since the `Sync` suffix has special meaning on many functions.  We've used "quiescence" in other circumstances where we're waiting for tranquility, though I'm not convinced that's a great word either. :smile:

Reply:
Depending on the final solution here, any aggressive file operation should also take place within a `Fiber`.